src/battle/battle_system.c

```diff
@@ -1084,3 +1088,218 @@ u32 BattleSystem_GetRandTemp(BattleSystem *bsys) {
 void BattleSystem_SetRandTemp(BattleSystem *bsys, u32 temp) {
     bsys->randTemp = temp;
 }
+
+void ov12_0223BDDC(BattleSystem *bsys, int battlerId, u8 data) {
+    if (!(bsys->battleSpecial & BATTLE_SPECIAL_RECORDED) && bsys->unk245C[battlerId] < 1024) {
+        sub_02030260(battlerId, bsys->unk245C[battlerId], data);
+        bsys->unk245C[battlerId]++;
+    }
+}
+
+BOOL ov12_0223BE0C(BattleSystem *bsys, int battlerId, u8 *data) {
+    BOOL ret;
+    *data = -1;
+    if ((bsys->battleSpecial & BATTLE_SPECIAL_RECORDED) && (bsys->unk245C[battlerId] < 1024)) {
+        *data = sub_0203027C(battlerId, bsys->unk245C[battlerId]);
+        bsys->unk245C[battlerId]++;
+        ret = FALSE;
+    } else if ((bsys->battleSpecial & BATTLE_SPECIAL_RECORDED) && bsys->unk245C[battlerId] >= 1024) {
+        ret = TRUE;
+    }
+    return ret;
+}
+
+u8 ov12_0223BE68(BattleSystem *bsys, u8 *buffer) {
+    u8 i;
+    u8 battlerId;
+    u8 index = 0;
+    
+    for (battlerId = 0; battlerId < bsys->maxBattlers; battlerId++) {
+        if (bsys->unk245C[battlerId] != bsys->unk244C[battlerId]) {
+            buffer[index++] = battlerId;
+            buffer[index++] = bsys->unk245C[battlerId] - bsys->unk244C[battlerId];
+            for (i = 0; i < bsys->unk245C[battlerId] - bsys->unk244C[battlerId]; i++) {
+                buffer[index++] = sub_0203027C(battlerId, bsys->unk244C[battlerId] + i);
+            }
+            bsys->unk244C[battlerId] = bsys->unk245C[battlerId];
+        }
+    }
+    
+    GF_ASSERT(index <= 28);
+    
+    return index;
+}
+
+static u8 ov12_0223BFB0(u8 *buffer, u8 *index, u16 *size);
```

Comment:
statics should really be at the top of the file

Reply:
u right